libflux/flux-core/src/semantic/formatter/tests.rs

```diff
@@ -0,0 +1,118 @@
+use crate::semantic::convert_source;
+use crate::semantic::formatter::format;
+use pretty_assertions::assert_eq;
+
+#[test]
+fn literals() {
+    let sem_pkg = convert_source(
+        r#"
+            a = "Hello, World!"
+            b = 12
+            c = 18.5
+            d = 12h
+            e = 2019-10-31T00:00:00Z
+            f = /server[01]/
+        "#,
+    )
+    .unwrap();
+
+    let got = format(&sem_pkg).unwrap();
+
+    let want = r#"package main
+a = "Hello, World!":string
+b = 12:int
+c = 18.5:float
+d = 043200000000000false:duration
```

Comment:
This formatting could be improved, can you make it so that it shows `12h:duration` instead of this? Maybe you could match what the AST formatter does... or is it that the representation of durations changes somehow between ASt and semantic graph?